Update Upgrading guide for OpenVox (#4769)
theforeman/foreman-documentation: guides/common/modules/proc_upgrade-puppet-agent-to-openvox-agent.adoc

:_mod-docs-content-type: PROCEDURE

[id="upgrade-puppet-agent-to-openvox-agent"]

> = Upgrade puppet-agent to openvox-agent

Reviewer: I'm not convinced we need a full blown procedure for a single step. Have you considered adding this as a sub-step to the list of post-upgrade tasks that's part of the server upgrade procedure?

Author: That is what I was considering, but the problem is this has to happen after the repos are enabled in `Synchronizing the new repositories`. Another (probably trivial) problem is that this step has to be performed on the hosts and not on the foreman server itself.

Reviewer: Oh, the dependency on syncing the repos is something I missed... okay then! Thanks for considering it :)

Reviewer: @ekohl Here is where we discussed the question of whether to have a full procedure or not.

[role="_abstract"]

> If you use puppet for host management, you have the option to upgrade `puppet-agent`, installed on your hosts, to `openvox-agent`.

Reviewer 2: ```suggestion
If you use Puppet for host management, you have the option to upgrade `puppet-agent`, installed on your hosts, to `openvox-agent`.
```

> If you use Puppet for host management, you have the option to upgrade `puppet-agent`, installed on your hosts, to `openvox-agent`.

Reviewer 3: Is "upgrade" the correct term? This sounds like we value OpenVox higher than Puppet.

Reviewer 2: It upgrades from version 7 to 8 as well so IMHO it is an upgrade.

Reviewer 3: Fair enough.

Author: We are doing a package update from `puppet-agent 7` to `openvox-agent 8`, should I mention the package versions explicitly?

Author: Apologies, missed that this was marked as resolved.

> If you use Puppet for host management, you have the option to upgrade `puppet-agent`, installed on your hosts, to `openvox-agent`.

Reviewer: You could simplify the wording a little bit and I also think it would be better to refer to "product names" (with version) rather than package names here. Using more generic terms seems like a better choice for an abstract:

```suggestion
If you use Puppet for host management, you can upgrade from Puppet agent 7 to OpenVox agent 8.
```

Reviewer: The heading should also be updated to match.

> 

Reviewer: The abstract says that the procedure is optional, but it's not clear when users should take these steps. How do they decide whether they want to upgrade to OpenVox agent or not? Can we add something like this to provide the additional context?

```suggestion
Puppet agent 7 remains compatible with OpenVox in {Project} as a legacy alternative, but {Team} recommends to upgrade to OpenVox agent 8.
For more information, see [link to https://docs.theforeman.org/3.18/Managing_Configurations_Puppet/index-satellite.html#openvox-versions-supported-for-integration-with-satellite].
```

.Prerequisites

> * *{project-client-name} OpenVox* repository is enabled on the host.

Reviewer 2: For upstream this will be wrong. Perhaps we should keep the procedure specific to Satellite by hiding it behind `ifdef` statements in the `master.adoc` files?
